File: Bridge/System/WeakReference.cs
using System.Runtime.Serialization;
using Bridge;

namespace System
{
    public class WeakReference : ISerializable
    {
        public WeakReference(object target)
            : this(target, false)
        {
        }

        public WeakReference(object target, bool trackResurrection)
        {
        }

        protected WeakReference(SerializationInfo info, StreamingContext context)
        {
        }

        public virtual bool IsAlive
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public virtual object Target
        {
            get
            {
                throw new NotImplementedException();
            }
            set
            {
                throw new NotImplementedException();
            }
        }

        public virtual bool TrackResurrection
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public virtual void GetObjectData(SerializationInfo info, StreamingContext context)
        {
            throw new NotImplementedException();
        }
    }

    public class WeakReference<T>
    {
        public WeakReference(T target)
        {
        }

        public void SetTarget(T target)
        {
            throw new NotImplementedException();
        }

        public bool TryGetTarget(out T target)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/weakReference.js
    Bridge.define("System.WeakReference$1", function (T) { return {
        ctors: {
            ctor: function (target) {
                this.$initialize();
            }
        },
        methods: {
            setTarget: function (target) {
                throw new System.NotImplementedException();
            },
            tryGetTarget: function (target) {
                throw new System.NotImplementedException();
            }
        }
    }; });

    Bridge.define("System.WeakReference", {
        inherits: [System.Runtime.Serialization.ISerializable],
        props: {
            IsAlive: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            Target: {
                get: function () {
                    throw new System.NotImplementedException();
                },
                set: function (value) {
                    throw new System.NotImplementedException();
                }
            },
            TrackResurrection: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        alias: ["getObjectData", "System$Runtime$Serialization$ISerializable$getObjectData"],
        ctors: {
            ctor: function (target) {
                System.WeakReference.$ctor1.call(this, target, false);
            },
            $ctor1: function (target, trackResurrection) {
                this.$initialize();
            },
            $ctor2: function (info, context) {
                this.$initialize();
            }
        },
        methods: {
            getObjectData: function (info, context) {
                throw new System.NotImplementedException();
            }
        }
    });
